### DISCUSSION TITLE
Flatpak PWA using a separate process/.desktop file

### DISCUSSION THREAD

**Discussion by cameronj86** [Floorp-Projects/Floorp#1099] [Feature Request]
Great to see the continued work on the PWA for Linux! Being able to start them from the app launcher is a much cleaner user experience, and definitely is easier than manually selecting it from the menu. 

I've played around w/ it for a few hours and did run into 1-2 minor inconveniences and wanted to provide some feedback and/or hear if anyone has any workarounds I'm not aware of: 

1. When I try to close Floorp, it automatically closes the PWA as well. It would be great if it could run as a separate window/process so that it could stand alone, even if it takes up more resources. That way, I can run things like a messaging PWA by itself.
2. The PWA inherits the Floorp Icon and is indistinguishable from the browser itself in alt+tab switcher (and the app launcher unless ungrouped). This could be due to it being the same process too? Or is there a way to assign a new icon via a .desktop file that is configurable a la Chrome PWAs?
3. I can pin a PWA to the application launcher bar, but once the app is closed, the PWA completely disappears. When the app is relaunched, it reappears in the app launcher. Not sure whether this is process or .desktop file related either.

Overall, though it's been a great upgrade, and I've been able to lean on PWA's more heavily since they're a lot more accessible thanks to them being a part of the app launcher!

**Comment by CutterKnife** [comment]
1. Perhaps this is a limitation of Mozilla's API. I think it would have worked the same on Windows.
2. https://github.com/Floorp-Projects/Floorp/issues/1102
